r1cs/src/gadget/binary.rs

```diff
@@ -0,0 +1,58 @@
+use crate::driver::CircuitDriver;
+use crate::{R1cs, Wire};
+use std::marker::PhantomData;
+
+#[derive(Clone)]
+pub struct BinaryAssignment<C: CircuitDriver>(Wire, PhantomData<C>);
```

Comment:
We can replace CircuitDriver with PrimeField

Reply:
Will fix it.

Reply:
We use R1cs<C>